editor/src/components/canvas/controls/namerowcrumbs.tsx

```diff
@@ -0,0 +1,104 @@
+import * as React from 'react'
+import { getOpenUtopiaJSXComponentsFromState } from '../../../components/editor/store/editor-state'
+import { useEditorState } from '../../../components/editor/store/store-hook'
+import { usePropControlledRef_DANGEROUS } from '../../../components/inspector/common/inspector-utils'
+import { InstancePath, TemplatePath } from '../../../core/shared/project-file-types'
+import { betterReactMemo, getControlStyles, SelectOption, Utils } from '../../../uuiui-deps'
+import * as TP from '../../../core/shared/template-path'
+import * as EditorActions from '../../editor/actions/action-creators'
+import { GridRow } from '../../../components/inspector/widgets/grid-row'
+import { PopupList } from '../../../uuiui'
+import { typeOptions } from '../../../components/inspector/sections/header-section/name-row'
+import { JSXElementName, jsxElementName } from '../../../core/shared/element-template'
+import { useNamesAndIconsSelectedViews } from '../../../components/inspector/common/name-and-icon-hook'
+
+export const NameRowCrumbs = () => {
+    const selectedViews = useEditorState(
+        (store) => store.editor.selectedViews,
+        'InspectorEntryPoint selectedViews',
+    )
+
+    const refElementsToTargetForUpdates = usePropControlledRef_DANGEROUS(
+        getElementsToTarget(selectedViews),
+    )
+
+    const { dispatch, jsxMetadataKILLME, rootComponents } = useEditorState((store) => {
+        return {
+            dispatch: store.dispatch,
+            jsxMetadataKILLME: store.editor.jsxMetadataKILLME,
+            rootComponents: getOpenUtopiaJSXComponentsFromState(store.editor),
+        }
+    }, 'InspectorContextProvider')
+
+    function getElementsToTarget(paths: Array<TemplatePath>): Array<InstancePath> {
+        let result: Array<InstancePath> = []
+        Utils.fastForEach(paths, (path) => {
+            // TODO Scene Implementation
+            if (!TP.isScenePath(path) && !TP.containsPath(path, result)) {
+                result.push(path)
+            }
+        })
+        return result
+    }
+
+    const onElementTypeChange = React.useCallback(
+        (value: JSXElementName) => {
+            const actions = refElementsToTargetForUpdates.current.map((path) => {
+                return EditorActions.updateJSXElementName(path, value)
+            })
+            dispatch(actions, 'everyone')
+        },
+        [dispatch, refElementsToTargetForUpdates],
```

Comment:
You don't need to include `refElementsToTargetForUpdates` here since it is a ref.

Reply:
unfortunately eslint doesn't know that, and it doesn't cause any problems if it is included. I've done this to pseudo-refs myself too

Reply:
Commit #a75137b addresses this